services/provider/server/server.go

```diff
@@ -146,7 +148,10 @@ func (s *OCSProviderServer) AcknowledgeOnboarding(ctx context.Context, req *pb.A
 		}
 		return nil, status.Errorf(codes.Internal, "Failed to update the storageConsumer. %v", err)
 	}
-
+	// create noobaa account CR
+	if err := s.consumerManager.CreateNoobaaAccount(ctx, req.StorageConsumerUUID); err != nil {
```

Comment:
if creation of NoobaaAccount isn't conditional why can't the whole operation be offloaded to StorageConsumer controller? I believe this is what Ohad mentioned to me once.

thinking about it, as a side-effect/requirement to support StorageConsumer a couple of Rook resources are created and likewise if NB is also a requirement the controller is a good place. Deletion could be done via OwnerRef and not in a stepped phase, wdyt?

Reply:
That change was done. resolving